_posts/2022-04-11-1.15-announcement.md

```diff
@@ -0,0 +1,375 @@
+can be activated. Users can define 
+alignment groups for which consuming from sources which are too far ahead from others 
+are paused. The ideal case for aligned watermarks is when there are two or more 
+sources that produce watermarks at a different speed and when the source has the same 
+parallelism as splits/shards/partitions.
+
+
+### SQL version upgrades
+
+The execution plan of SQL queries and its resulting topology is based on optimization 
+rules and a cost model. This means even minimal changes could introduce a completely 
+different topology. This dynamic setting makes guaranteeing snapshot compatibility 
+really hard across Flink versions. In the efforts of 1.15, the community put the focus 
+on keeping the same query up and running even after upgrades. At the core of SQL 
+upgrades are JSON plans 
+([for now we have only documentation in our JavaDoc, while we are still working on updating the documentation](https://nightlies.apache.org/flink/flink-docs-release-1.15/api/java/org/apache/flink/table/api/CompiledPlan.html)), 
+they have been introduced for 
+internal use already in previous releases and will now be exposed. Both the Table API 
+and SQL will provide a way to compile and execute a plan which guarantees the same 
+topology throughout versions. This feature will be released as an experimental MVP. 
+Users who want to give it a try already can create a JSON plan that can then be used 
+to restore a Flink Job based on the old operator structure. The first real upgrade 
+will then happen in Flink 1.16.
+
+
+### Changelog state backend
+
+From Flink 1.15, we have the MVP feature of 
+[changelog state](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/ops/state/state_backends/#enabling-changelog)
+backend, which aims at 
+making checkpoint intervals shorter with following advantages:
+
+1. Less work on recovery: The more frequently a checkpoint is taken, the fewer events 
+   need to be re-processed after recovery.
+2. Lower latency for transactional sinks: Transactional sinks commit on checkpoints, 
+   so faster checkpoints mean more frequent commits.
+3. More predictable checkpoint intervals: Currently the length of the checkpoint mainly
+   depends on the size of the artifacts that need to be persisted on the checkpoint 
+   storage, by keeping the size small it becomes more predictable.
+
+This work introduced in Flink 1.15 helps achieve the above advantages by continuously 
+persisting state changes on a non-volatile storage, while performing materialization 
+in the background.
+
+
+### Adaptive batch scheduler
+
+In 1.15, we introduced a new scheduler to Apache Flink: the 
+[Adaptive Batch Scheduler](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/elastic_scaling/#adaptive-batch-scheduler). 
+The new scheduler can automatically decide parallelisms of job vertices for batch jobs, 
+according to the size of data volume each vertex needs to process.
+
```

Comment:
```suggestion

### Repeatable Cleanup

In previous releases of Flink, cleaning up job-related artifacts was done only once which might have resulted in abandoned artifacts in case of an error. In this version, Flink will try to run the cleanup again to avoid leaving artifacts behind. This retry mechanism runs until it was successful, by default. Users can change this behavior by configuring the [repeatable cleanup options](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/config/#retryable-cleanup). Disabling the retry strategy will lead to Flink behaving like in previous releases.

There is still work in progress around cleaning up Checkpoints, which is covered by [FLINK-26606](https://issues.apache.org/jira/browse/FLINK-26606).
```
Just in case this was missed accidentally. I think, the repeatable cleanup should end up in the release blog post as well. Feel free to adapt my proposal for the paragraph.

Reply:
That's alright - if you guys considered it to be less important. I'm fine with keeping it out of the blog post. 👍 
My only concern was that users might not be aware of Flink's default behavior chaning in that regards... The job cleanup is not visualized in the UI, yet.